src/spikeinterface/core/binaryrecordingextractor.py

```diff
@@ -148,6 +148,18 @@ def get_binary_description(self):
         )
         return d
 
+    def __del__(self):
+        """
+        Ensures that all segment resources are properly cleaned up when this recording extractor is deleted.
+        Closes any open file handles in the recording segments.
+        """
+        # Close all recording segments
+        if hasattr(self, "_recording_segments"):
+            for segment in self._recording_segments:
+                # This will trigger the __del__ method of the BinaryRecordingSegment
+                # which will close the file handle
+                del segment
```

Comment:
Why not moving this to the `BaseRecording`? I think deleting segments is a safe thing to do for any object! We could do it for sortings as well!

Reply:
I can do this but I would like to get this through and do the general change in another PR. In case the general change fails or there is something else we are missing at least we have this feature already published.

would you be OK with that?

Reply:
yep. Merging as is then :)